community/kernel/src/main/java/org/neo4j/kernel/impl/store/TemporalType.java

```diff
@@ -0,0 +1,452 @@
+            },
+    TEMPORAL_LOCAL_DATE_TIME( 3, "LocalDateTime" )
+            {
+                @Override
+                public Value decodeForTemporal( long[] valueBlocks, int offset )
+                {
+                    long nanoOfSecond = valueBlocks[offset] >>> 32;
+                    long epochSecond = valueBlocks[1 + offset];
+                    return LocalDateTimeValue.localDateTime( epochSecond, nanoOfSecond );
+                }
+
+                @Override
+                public int calculateNumberOfBlocksUsedForTemporal( long firstBlock )
+                {
+                    return 2;
+                }
+
+                @Override
+                public ArrayValue decodeArray( TemporalHeader header, byte[] data )
+                {
+                    throw new UnsupportedOperationException( "Add support for arrays" );
+                }
+            },
+    TEMPORAL_TIME( 4, "Time" )
+            {
+                @Override
+                public Value decodeForTemporal( long[] valueBlocks, int offset )
+                {
+                    int secondOffset = (int) (valueBlocks[offset] >>> 32);
+                    long nanoOfDay = valueBlocks[1 + offset];
+                    return TimeValue.time( nanoOfDay, ZoneOffset.ofTotalSeconds( secondOffset ) );
+                }
+
+                @Override
+                public int calculateNumberOfBlocksUsedForTemporal( long firstBlock )
+                {
+                    return 2;
+                }
+
+                @Override
+                public ArrayValue decodeArray( TemporalHeader header, byte[] data )
+                {
+                    throw new UnsupportedOperationException( "Add support for arrays" );
+                }
+            },
+    TEMPORAL_DATE_TIME( 5, "DateTime" )
+            {
+                @Override
+                public Value decodeForTemporal( long[] valueBlocks, int offset )
+                {
+                    if ( storingZoneOffset( valueBlocks[0] ) )
+                    {
+                        int nanoOfSecond = (int) (valueBlocks[offset] >>> 33);
+                        long epochSecond = valueBlocks[1 + offset];
+                        int secondOffset = (int) valueBlocks[2 + offset];
+                        return DateTimeValue.datetime( epochSecond, nanoOfSecond, ZoneOffset.ofTotalSeconds( secondOffset ) );
+                    }
+                    else
+                    {
+                        throw new UnsupportedOperationException( "..." );
+                    }
+                }
+
+                @Override
+                public int calculateNumberOfBlocksUsedForTemporal( long firstBlock )
+                {
+                    if ( storingZoneOffset( firstBlock ) )
+                    {
+                        return 3;
+                    }
+                    else
+                    {
+                        // TODO proper number
+                        return 3;
```

Comment:
Then what is a proper number?

Reply:
We don't know yet. This will also be figured out when we implement the format for DateTimes with ZoneIds.